Google Colab Demo

albertotono opened the pull request:
Thanks a lot for this great project, I went ahead and share a Google Colab Demo so people will be able to test it from there. I hope it will help.

ShashwatNigam99 commented:
@albertotono This notebook copy requires you to share it publicly. Right now people need to request access. And yeah, great idea to make the ready to use version available for everyone; I did have a tough time getting it running for the first time on colab (matching the torch, torchvision versions to colab's python 3.6.9 and cuda 10.0 etc, and then changing the number of GPUs to 1, also changing the batch-size from 16 to 8 because meshrcnn recommends 8GPU training, running on 1 GPU will give you out of memory error) .

@gkioxari Thanks for helping out whenever I was stuck!

albertotono commented:
Thanks @ShashwatNigam99 I made it accessible to everyone now that you agreed in sharing. Thanks for the details. Yes it is an amazing work that you did.  I added the option to upload files in an easier way and I added instructions where to change the filenames and the download option.  I hope it will help the community.

gkioxari commented:
Awesome!! Thank you @albertotono and @ShashwatNigam99! I don't use Colab so I really appreciate your work in making it accessible there! I am merging the PR!  

P.S. It's nice to see strangers come together and help each other :)

albertotono commented:
Thanks @gkioxari! 

yes @ShashwatNigam99 and I met thanks to MeshRCNN and Pytorch3D. Our passion for the topic and this amazing open-source project helped a lot. We are working on a research paper together now. Thanks again so much for the help and availability. 

We will be around if someone else will need further help with Colab. 

Thanks

manannarang commented:
Thank you so much for this

On Mon, Oct 26, 2020, 06:52 Facebook Community Bot <notifications@github.com>
wrote:

> @gkioxari <https://github.com/gkioxari> merged this pull request in
> 1ce61f3
> <https://github.com/facebookresearch/meshrcnn/commit/1ce61f3ccda8dac2ff2c426a14d678e99016277a>
> .
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebookresearch/meshrcnn/pull/69#issuecomment-716252121>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMTOUFT4DWS7OKAU24CPWKTSMTFMVANCNFSM4SL4NB4A>
> .
>

manannarang commented:
I am getting this error while running kindly help me out.
[image: Screenshot from 2020-10-26 09-51-31.png]
[image: Screenshot from 2020-10-26 09-51-44.png]


On Mon, Oct 26, 2020 at 8:04 AM MXANAXN <iammanannarang@gmail.com> wrote:

> Thank you so much for this
>
> On Mon, Oct 26, 2020, 06:52 Facebook Community Bot <
> notifications@github.com> wrote:
>
>> @gkioxari <https://github.com/gkioxari> merged this pull request in
>> 1ce61f3
>> <https://github.com/facebookresearch/meshrcnn/commit/1ce61f3ccda8dac2ff2c426a14d678e99016277a>
>> .
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/facebookresearch/meshrcnn/pull/69#issuecomment-716252121>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AMTOUFT4DWS7OKAU24CPWKTSMTFMVANCNFSM4SL4NB4A>
>> .
>>
>

albertotono commented:
Dear @manannarang the screenshots don't display correctly, 

the only thing you need to do is load your image of couch, table, desk, sofa, etc,  following the instructions. 
It should take a while at the begging to load all the libraries for the first time. 

you can or name the picture accordingly or change the filepath. 

Let me know if you need further help.

manannarang commented:
![Screenshot from 2020-10-26 09-51-31](https://user-images.githubusercontent.com/52881942/97135783-6223ea00-1777-11eb-852f-d05d6c2edd04.png)
![Screenshot from 2020-10-26 09-51-44](https://user-images.githubusercontent.com/52881942/97135784-63551700-1777-11eb-883c-61fea3c22a74.png)

albertotono commented:
on the **maskname** you need to change the name based on the output mask. 
The code generates the prediction based on the object recognized. 


so for example if it recognize that the object is a sofa it will call:
maskname = '0_mask_**sofa**_1.000.png'

what is the object in your case? 
just change the part in bold based on your object. 

you should try to navigate the folder structure on your left. 
![image](https://user-images.githubusercontent.com/11828200/97136111-7b914b80-170f-11eb-95a3-9ece63cb963b.png)
this part, after you run the code and update the images will get populated.

manannarang commented:
![img](https://user-images.githubusercontent.com/52881942/97136526-5802eb00-1779-11eb-90f0-095349d705d9.png)


This is the image, it's a solar panel.

albertotono commented:
yes, it will output 1 of the 9 categories in which it has been trained. 

You need to check the file name on the folder structure and change accordingly. 

Don't expect huge results, since there were no solar panel in the training set.

manannarang commented:
How to train it then?

albertotono commented:
This Colab is just to run the demo.py as described in the title. 

To train the model, you will need to follow the instructions here  https://github.com/facebookresearch/meshrcnn/blob/master/INSTRUCTIONS_PIX3D.md 

It is recommended to train the model with 8 GPU Tesla V100. 

Were you able to complete the demo successfully?!

manannarang commented:
I am using Jetson Xavier to train it, the demo uses Mask R CNN for now. This is what I was getting when I open the directory.

![Screenshot from 2020-10-26 11-17-53](https://user-images.githubusercontent.com/52881942/97138064-53d8cc80-177d-11eb-9dd4-9c4d654f69ac.png)

albertotono commented:
Everything is under the **content** folder when you run a code in colab. 

I create a tutorial for you, I hope it will help @manannarang 

https://www.youtube.com/watch?v=DmCh7ADpXnE

post here your results

manannarang commented:
I think I need to train the data first because it is unable to identify it right now. I have images of the panel in various angles with 2d annotations with the json file. Will I be able to run it using the NVIDA Jetson Xavier? If so, How?
![Screenshot from 2020-10-26 14-02-20](https://user-images.githubusercontent.com/52881942/97150370-07e55200-1794-11eb-8f73-aeedf6d1d99c.png)

manannarang commented:
Hi I ran the code but the result wasn't what I was expecting.. 

![Screenshot from 2020-10-26 18-28-27](https://user-images.githubusercontent.com/52881942/97175378-6ffa5f00-17b9-11eb-96e1-db99d631054e.png)
![Screenshot from 2020-10-26 18-23-41](https://user-images.githubusercontent.com/52881942/97175388-72f54f80-17b9-11eb-8637-d6c4e26cb5ce.png)
![Screenshot from 2020-10-26 18-25-14](https://user-images.githubusercontent.com/52881942/97175394-74bf1300-17b9-11eb-97e4-49c03592c521.png)
![Screenshot from 2020-10-26 18-27-02](https://user-images.githubusercontent.com/52881942/97175401-7688d680-17b9-11eb-8c5f-291fca706b9a.png)

manannarang commented:
I tried doing the training but I'm having trouble running it. Can you help me out by making another demo? It would be great! @albertotono @gkioxari

albertotono commented:
Hi Manan,

Sure, I will more than welcome to help, if I can.

Are you trying to train the MeshRCNN  model with a single GPU? What dataset
are you using ?

( keep in mind that it is not recommended, I guess your is more an exercise
than a research)





On Wed, Oct 28, 2020 at 9:14 PM Manan Narang <notifications@github.com>
wrote:

> I tried doing the training but I'm having trouble running it. Can you help
> me out by making another demo? It would be great! @albertotono
> <https://github.com/albertotono> @gkioxari <https://github.com/gkioxari>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebookresearch/meshrcnn/pull/69#issuecomment-718348084>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2HX2GZD7IGJJAZUVDIWULSNDTYVANCNFSM4SL4NB4A>
> .
>
-- 

*Eng. Alberto Tono, LEED AP BD+C*

us *+14158672390 *

*Facebook <https://www.facebook.com/alberto.tono>** | **Twitter*
<https://twitter.com/albertotono3?lang=it>  | *LinkedIn
<https://it.linkedin.com/in/albertotono3> | Github
<https://github.com/albertotono>*

manannarang commented:
I am using my HP laptop with Ubuntu and a Jetson Xavier to train it. I have multiple photos of the solar panel from different angles. @albertotono

albertotono commented:
What is your goal, not sure I am understanding what you are trying to
achieve. Why do you think MeshRCNN is good for your goal?

On Wed, Oct 28, 2020 at 9:22 PM Manan Narang <notifications@github.com>
wrote:

> I am using my HP laptop with Ubuntu and a Jetson Xavier to train it. I
> have multiple photos of the solar panel from different angles.
> @albertotono <https://github.com/albertotono>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebookresearch/meshrcnn/pull/69#issuecomment-718350385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2HX2HVKCXJXSFUR7FQHT3SNDUZFANCNFSM4SL4NB4A>
> .
>
-- 

*Eng. Alberto Tono, LEED AP BD+C*

us *+14158672390 *

*Facebook <https://www.facebook.com/alberto.tono>** | **Twitter*
<https://twitter.com/albertotono3?lang=it>  | *LinkedIn
<https://it.linkedin.com/in/albertotono3> | Github
<https://github.com/albertotono>*

manannarang commented:
I'm trying to build 3d objects from photos so I need to add different objects by training. On a broader note I'm working on computer vision on this project.

manannarang commented:
Any update?

manannarang commented:
@albertotono @gkioxari

albertotono commented:
Sorry Manan, I don't see any questions here.

If you are asking if MeshRCNN is good for your specific use case I would say no. Use a simple image recognition. Take pre-trained model with imagenet, fine-tune with your solar pannel and then add a way to give the 3d object after the recognition. 

Have fun.
